### PR TITLE
Switch the TSV Zeek logs to be UTF8 by default.

### PR DESCRIPTION
**Pull request by sethhall**
This will give us equivalent behavior with the json logs.  The only thing I'm not sure of, is if the ascii tab character is a valid utf-8 byte?  If the tab character can be a utf8 byte then I'm not sure about this whole branch.

There is a paired zeek-testing branch for some updates there.

### CODE REVIEW

**Comment by sethhall** [comment]
I'm not sure what to do about the failing test.  It actually passed when I ran it locally but it looks like the code that stashes artifacts is freaking out about the newly discovered utf8 characters (which do seem to be some garbage that just happen to be valid utf8 characters).

File: NEWS [hunk 1]
```diff
@@ -49,6 +49,8 @@ New Functionality
 Changed Functionality
 ---------------------
 
+- The traditional TSV Zeek logs are now valid UTF8 by default.
```
Comment (jsiwek) [inline NEWS:52]: May help to explicitly mention it's possible to revert to old behavior via `LogAscii::enable_utf_8`.

**Review by jsiwek — COMMENTED** [review]
> The only thing I'm not sure of, is if the ascii tab character is a valid utf-8 byte?

Not sure if it's what you mean, but the binary representation of a "tab" in ASCII would be the same as it is in UTF-8 and so is "valid" (same goes for all 128 ASCII characters), no ?

> If the tab character can be a utf8 byte then I'm not sure about this whole branch.

What's the thinking here?  Is it about "tab" specifically?

There was care taken about (still) having the logging logic search for and escape any meaningful-meta characters/sequences (field separators and empty/unset value indicators) it finds within the actual log content regardless of ASCII versus UTF-8 output behavior, so don't think there's an issue with UTF-8 output somehow being more ambiguous to interpret.  You could check if test cases like this work like you expect:

https://github.com/zeek/zeek/blob/07bad2d40e0f9e0bb24da5d59955872189b1709e/testing/btest/scripts/base/frameworks/logging/ascii-utf8-enabled-control-characters.zeek#L18-L20

https://github.com/zeek/zeek/blob/07bad2d40e0f9e0bb24da5d59955872189b1709e/testing/btest/Baseline/scripts.base.frameworks.logging.ascii-utf8-enabled-control-characters/test.log#L8-L10

> I'm not sure what to do about the failing test. It actually passed when I ran it locally but it looks like the code that stashes artifacts is freaking out about the newly discovered utf8 characters

Think the primary problem is more that this doesn't update the `testing/external/commit-hash.zeek-testing*` files which help tell CI it needs to run against associated commits in the `external/` test suites that have updated baselines.  Want to do that?

**Comment by 0xxon** [comment]
Just checking if you still want to do this, @sethhall.

**Comment by rsmmr** [comment]
What's the state here?

**Comment by timwoj** [comment]
`{"ts":1258589289.945278,"uid":"CYtsEY110tlFGFeyRe","id.orig_h":"192.168.1.105","id.orig_p":49613,"id.resp_h":\
"65.55.53.190","id.resp_p":80,"trans_depth":1,"method":"GET","host":"watson.microsoft.com","uri":"/StageOne/G\
eneric/PCA2/mdd_1_3_exe/1_3_0_0/mdd/unknown/蓓ʆ￾￿촑瞙�瞙벹ל1-/200/ڠ.htm?LCID=1033&OS=6.0.6002.2.00010100.2.0\
.6.18005","version":"1.1","user_agent":"MSDW","request_body_len":0,"response_body_len":1635,"status_code":404\
,"status_msg":"Not Found","tags":[],"resp_fuids":["F8p4Qn20NOvGmbt434"],"resp_mime_types":["text/html"],"clie\
nt_header_names":["USER-AGENT","HOST","CONNECTION"],"uri_vars":["/StageOne/Generic/PCA2/mdd_1_3_exe/1_3_0_0/m\
dd/unknown/%E8%93%93%CA%86%EF%BF%BE%EF%BF%BF%EC%B4%91%E7%9E%99%EF%BF%BD%E7%9E%99%EB%B2%B9%D7%9C%EE%8B%A0%DA%A\
0/200/-1.htm?LCID","OS"]}`

I looked at the failing test, and there's a big question about how this should even be logged. The escaped characters in the `uri_vars` variable are most likely not intended to be character data. Converting them into UTF-8 to be displayed is probably not the right thing to be doing here but I'm not sure what the right thing is.

**Comment by timwoj** [comment]
From that byte string:

```
coregeek:~ tim$ printf "\xE8\x93\x93\xCA\x86\xEF\xBF\xBE\xEF\xBF\xBF\xEC\xB4\x91\xE7\x9E\x99\xEF\xBF\xBD\xE7\x9E\x99\xEB\xB2\xB9\xD7\x9C\xEE\x8B\xA0\xDA\xA0" | uniname -r
No LINES variable in environment so unable to determine lines per page.
Using default of 24.
character  byte       UTF-32   encoded as     glyph   range                                   name
        0          0  0084D3   E8 93 93       蓓      CJK Unified Ideographs                  No character name available
        1          3  000286   CA 86          ʆ      IPA Extensions                          LATIN SMALL LETTER ESH WITH CURL
        2          5  00FFFE   EF BF BE       ￾      Specials                                Undefined
        3          8  00FFFF   EF BF BF       ￿      Specials                                Undefined
        4         11  00CD11   EC B4 91       촑      Hangul Syllables                        HANGUL SYLLABLE COLG
        5         14  007799   E7 9E 99       瞙      CJK Unified Ideographs                  No character name available
        6         17  00FFFD   EF BF BD       �      Specials                                REPLACEMENT CHARACTER
        7         20  007799   E7 9E 99       瞙      CJK Unified Ideographs                  No character name available
        8         23  00BCB9   EB B2 B9       벹      Hangul Syllables                        HANGUL SYLLABLE BET
        9         26  0005DC   D7 9C          ל      Hebrew                                  HEBREW LETTER LAMED
       10         28  00E2E0   EE 8B A0             Private Use Area - Conlang Xaini        No character name available
       11         31  0006A0   DA A0          ڠ      Arabic                                  ARABIC LETTER AIN WITH THREE DOTS ABOVE
```

**Comment by timwoj** [comment]
The bytes are all clearly valid unicode in some way or another, but the decoded string makes zero sense. Should Zeek move to an all-or-nothing approach when decoding UTF-8? If it encounters something that doesn't make sense, the entire string should be escaped? We clearly need to explicitly handle things like the Specials and Private Use Areas as invalid characters.

**Comment by rsmmr** [comment]
Yeah, let me try a proposal:

1. Check if data looks & smells like a UTF-8 string, meaning (1) only valid UTF8 code points, and (2) no codepoint decodes to something "really weird" (like  Private Use Areas, Special, control characters).
2. If yes: fully decode as UTF-8 and log as such
3. If no: do not decode anything and log as we currently do for strings: keep what's printable in ASCII and `\xXX` everything else

That means that `\xXX` would only appear in non-UTF8 strings.

If somebody doesn't like this at all, they can turn off UTF8 through the global/filter options.

**Comment by rsmmr** [comment]
> Yeah, let me try a proposal:

Does that sound right?

**Comment by timwoj** [comment]
The above proposal is implemented in #1719

**Comment by timwoj** [comment]
I rebased this on master now that the change to fix escaping of broken UTF8 data has landed. I also fixed the one comment from Jon about NEWS mentioning how to revert to the previous behavior.

**Comment by timwoj** [comment]
Ok I've fixed everything with not escaping properly through the Ascii logger. Pushing this back to be a PR ready for review.

File: scripts/base/frameworks/logging/writers/ascii.zeek [hunk 2]
```diff
@@ -38,7 +38,7 @@ export {
 	## written into logs.
 	##
 	## This option is also available as a per-filter ``$config`` option.
-	const enable_utf_8 = F &redef;
+	const enable_utf_8 = T &redef;
```
Comment (rsmmr) [inline scripts/base/frameworks/logging/writers/ascii.zeek:41]: Nit: we write UTF8 elsewhere, so would suggest `enable_utf8`.

**Review by rsmmr — APPROVED** [review]

File: scripts/base/frameworks/logging/writers/ascii.zeek (hunk 2 above)
Comment (timwoj) [reply scripts/base/frameworks/logging/writers/ascii.zeek:41]: That's an existing option so I'm somewhat leery to change it.

**Pull request merged**